what4/src/What4/SWord.hs

```diff
@@ -314,8 +314,11 @@ bvSliceBE :: forall sym. IsExprBuilder sym => sym
   -> Integer
   -- ^ Starting index, from 0 as most significant bit
   -> Integer
-  -- ^ Number of bits to take (must be > 0)
-  -> SWord sym -> IO (SWord sym)
+  -- ^ Number of bits to take
+  -> SWord sym
+  -> IO (SWord sym)
+bvSliceBE _ _m 0 _  = pure ZBV
+bvSliceBE _ _ _ ZBV = return ZBV
```

Comment:
Allowing a slice from a ZBV seems invalid, and it doesn't agree with the implementation of `bvAtLE` and `bvAtBE` or with a slice of a non ZBV with invalid index or length... although it does agree with `bvSetBE` and `bvSetLE`